packages/opentelemetry-instrumentation-base/src/instrumentation-base.ts

```diff
@@ -0,0 +1,89 @@
+import { Meter, MeterProvider, metrics, trace, Tracer, TracerProvider } from "@opentelemetry/api";
+import * as path from "path";
+import * as RequireInTheMiddle from "require-in-the-middle";
+import * as semver from "semver";
+
+export abstract class Instrumentation {
+    protected abstract module: ModuleDefinition | ModuleDefinition[];
+    protected abstract patch<T>(exports: T, name: string, baseDir?: string): T;
+    protected abstract unpatch(): void;
+
+    private _hook: RequireInTheMiddle.Hooked = { unhook: () => { } };
+
+    private _tracer: Tracer;
+    private _meter: Meter;
+
+    constructor(protected readonly instrumentationName: string, protected readonly instrumentationVersion?: string) {
+        this._tracer = trace.getTracer(instrumentationName, instrumentationVersion);
+        this._meter = metrics.getMeter(instrumentationName, instrumentationVersion);
+    }
+
+    public enable() {
+        this._hook = RequireInTheMiddle(this._getModuleNames(), this._onRequire);
+    }
+
+    public disable() {
+        this._hook.unhook();
+        this.unpatch();
+    } 
+
+    public setTracerProvider(tracerProvider: TracerProvider) {
+        this._tracer = tracerProvider.getTracer(this.instrumentationName, this.instrumentationVersion);
+    }
+
+    public setMeterProvider(meterProvider: MeterProvider) {
+        this._meter =  meterProvider.getMeter(this.instrumentationName, this.instrumentationVersion);
+    }
+
+    protected get tracer(): Tracer {
```

Comment:
we are basically not using get/set approach in our sdk so would rather have `getTracer`, `getMeter` to just unify all ,  WDYT ?